hw1

matveeva-s opened the pull request:
Done :(

matveeva-s committed:
hw1

martinkomitsky reviewed (changes requested):
1. Не знаю, каким образом, но у тебя образовался конфликт, нужно его исправить в ветке путем подливки мастер ветки: `git pull origin master` и далее смотреть, что там сломалось.
2. Вынести из js файла все стили в css файл, в старом месте через js просто назначить css класс целевому div
3. Выполнить последний пункт из ДЗ про линтер
4. Добавить скрипт в package.json для запуска линтера
5. Запустить линтер, исправить ошибки, которые он покажет

matveeva-s committed:
fixed homework

martinkomitsky commented:
@matveeva-s все еще есть конфликт

matveeva-s commented:
При вызове git pull origin master показывает следующее: 
CONFLICT (modify/delete): src/lib/test.js deleted in d499ffe0f7abfdc641ad1d8043fd91c669d8b7b6 and modified in HEAD. Version HEAD of src/lib/test.js left in tree.
Removing src/index.js
CONFLICT (modify/delete): package.json deleted in d499ffe0f7abfdc641ad1d8043fd91c669d8b7b6 and modified in HEAD. Version HEAD of package.json left in tree.
CONFLICT (modify/delete): index.html deleted in d499ffe0f7abfdc641ad1d8043fd91c669d8b7b6 and modified in HEAD. Version HEAD of index.html left in tree.
Вообще я изначально очень накосячила с созданием ветки, из-за чего, думаю, проблемы могут быть в большом и непредсказуемом количестве. Спросила на последней лекции у Бориса, он предложил просто снести репозиторий и создать новый, если Вы за, то я так и сделаю.

martinkomitsky commented:
>При вызове git pull origin master показывает следующее:

Это и есть конфликт. Нужно зайти в каждый из этих файлов и исправить содержимое так, как они должны выглядеть к сдаче. 
После того, как исправишь эти файлы, надо будет написать следующее:
```
git add -A
git commit
git push origin HEAD
```
Ничего сносить не надо.

matveeva-s committed:
Merge branch 'master' of https://github.com/matveeva-s/2018-FS-11-Frontend-Matveeva into firstHW

martinkomitsky commented on src/lib/test.js:
```diff
@@ -0,0 +1,8 @@
+	div.classList.add("mydiv");
+	document.body.appendChild(div);
+};
+
```
Тут стало лучше, но, почему-то, я не вижу файла, откуда эта функция вызывается(

martinkomitsky reviewed (changes requested):
![image](https://user-images.githubusercontent.com/14847018/46945685-1f09e900-d07e-11e8-8a54-41d83f9fa0c5.png)
1. Ветка с ДЗ стала главнее матера)) Вернуть мастер на место, эта ветка должна быть default
2. В ветке присутствует сабмодуль, причем в самого себя:
![image](https://user-images.githubusercontent.com/14847018/46945715-3ea11180-d07e-11e8-83ff-ae84e5ece9f4.png). Его бы тоже неплохо удалить, он тут совсем не нужен
3. Мне кажется, что потерялся файл `index.js` в `./src/, где импортируется функция say и вызывается.
4. В package.json необходимо добавить 2 скрипта: `lint`, который будет проверять все имеющиеся js (передать на вход утилиты путь ./src/) и `build`, который будет собирать проект. Можно еще добавить каких-нибудь `start` для своего удобства.
5. После пункта 4 выполнить `npm run lint` и исправить все ошибки, на которые он указывает

martinkomitsky commented on src/lib/test.js:
```diff
@@ -0,0 +1,8 @@
+const say = function() {
+	var div = document.createElement('div');
+	div.innerHTML = "<strong>Hello teacher</strong>, have a nice day :)";
```
Тут будут ошибки от eslint, он скажет их исправить

martinkomitsky commented on package.json:
```diff
@@ -0,0 +1,36 @@
+  "description": "-",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Не используется, удалить

matveeva-s committed:
update

matveeva-s committed:
update

matveeva-s commented:
Так:
1) Ветка master теперь default
2) Субмодуль удален
3) index.js снова на своем месте (он почему-то лежал в субмодуле)
4) Добавила в поле scripts package.json-а 'build' и 'lint'
 Это из хорошего, а из того что не получилось:
 Вообще никак не запускается eslint:
1) Вручную через команду ./node_modules/.bin/eslint ./src/lib/test.js консоль просто ничего не выдает
2) Через npm run lint выдает следующее (возможно конечно я неправильно написала скрипт lint, но согласно гуглу это пишется так):
> test@1.0.0 lint /home/svetlana/front
> eslint ./src/index.js && ./node_modules/.bin/eslint ./src/lib/test.js

martinkomitsky commented on package.json:
```diff
@@ -0,0 +1,41 @@
+  "scripts": {
+    "build": "webpack -p",
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "lint": "./node_modules/.bin/eslint ./src/index.js && ./node_modules/.bin/eslint ./src/lib/test.js"
```
Тут неправильно) если у тебя будет 500 файлов, ты их ручками напишешь тут?))
укажи просто путь `./src/ и прогони тест еще раз

martinkomitsky commented on package.json:
```diff
@@ -0,0 +1,41 @@
+  "main": "index.js",
+  "scripts": {
+    "build": "webpack -p",
+    "test": "echo \"Error: no test specified\" && exit 1",
```
а эта строчка не нужна

matveeva-s committed:
update

matveeva-s committed:
final update

martinkomitsky committed:
eslint config

matveeva-s committed:
final final commit

matveeva-s committed:
:Merge branch 'firstHW' of https://github.com/matveeva-s/2018-FS-11-Frontend-Matveeva into firstHW

matveeva-s committed:
update

matveeva-s committed:
fixed .eslintrc.json and package.json

martinkomitsky commented:
10, наконец-то
